### PR TITLE
Porting the plugin to JasperReports 6.11.0

### PR DESCRIPTION
**Pull request by amchavan**
Three commits, the first one being the core one.
The second commit includes some optional code improvements (largely what IntelliJ thinks are improvements)
The third commit upgrades most test reports to 6.11.0 to avoid a compilation warning.

Hope this helps! We at www.eso.org would love to use this plugin with the latest JasperReports version.

### CODE REVIEW

File: src/test/resources/exampleFolders/sampleReports/AddressesReport.jrxml [hunk 1]
```diff
@@ -1,137 +1,136 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sepe81) [inline src/test/resources/exampleFolders/sampleReports/AddressesReport.jrxml]: Maybe for reasons of clarity and comprehensibility (git history) it's better to revert those formatting changes?

File: src/main/java/com/alexnederlof/jasperreport/CompileTask.java [hunk 2]
```diff
@@ -55,24 +55,13 @@ public CompileTask(File source, File destination, Log log, boolean verbose) {
      */
     @Override
     public Void call() throws Exception {
-        OutputStream out = null;
-        InputStream in = null;
-        try {
-            out = new FileOutputStream(destination);
-            in = new FileInputStream(source);
+        try (OutputStream out = new FileOutputStream(destination); InputStream in = new FileInputStream(source)) {
```
Comment (sepe81) [inline src/main/java/com/alexnederlof/jasperreport/CompileTask.java]: LGTM

File: src/test/resources/exampleFolders/sampleReports/AddressesReport.jrxml (hunk 1 above)
Comment (amchavan) [reply src/test/resources/exampleFolders/sampleReports/AddressesReport.jrxml]: Indeed! (But JasperSoft Studio thinks it knows better :-) 
I reverted the reformatting and pushed a new version of the file that only includes compatibility changes.

File: src/main/java/com/alexnederlof/jasperreport/CompileTask.java (hunk 2 above)
Comment (amchavan) [reply src/main/java/com/alexnederlof/jasperreport/CompileTask.java]: I had to look that up, I'm new to gitHub. If that means Looks Good To Me: thanks!

File: src/main/java/com/alexnederlof/jasperreport/CompileTask.java [hunk 3]
```diff
@@ -55,24 +55,13 @@ public CompileTask(File source, File destination, Log log, boolean verbose) {
      */
```
Comment (rotatomel) [inline src/main/java/com/alexnederlof/jasperreport/CompileTask.java]: Probably very nitpick, but in the javadoc here it says that this method will 

> skip the file if it doesn't have the right extension
Does it still makes sense? Why don't just remove this "documentation" at all.

Comment (amchavan) [reply src/main/java/com/alexnederlof/jasperreport/CompileTask.java]: It doesn't – removed.

**Comment by amchavan** [comment]
Thanks for pointing out those warnings – fixed.

**Comment by amchavan** [comment]
Added two more commits to this PR.

**Comment by sepe81** [comment]
@alexnederlof Could this be merged?

**Comment by amchavan** [comment]
Yes, can this be merged now? Is this good enough for releasing a new version of the plugin?

**Comment by alexnederlof** [comment]
sorry for the delay, it has been released now!

**Comment by amchavan** [comment]
Great! Thank you all for your help.

Maurizio

**Pull request merged**